rpi/Animal/lib/Animal.pm

```diff
@@ -29,107 +31,103 @@ Perhaps a little code snippet.
     my $foo = Animal->new();
     ...
 
-=head1 EXPORT
-
-A list of functions that can be exported.  You can delete this section
-if you don't export anything, such as for a purely object-oriented module.
-
 =head1 SUBROUTINES/METHODS
 
-=head2 function1
+=head2 named
 
-=cut
+Create an animal with a name
 
-sub speaks_english {
-  return 0;
-}
+=cut
 
-=head1 AUTHOR
+sub named {
+  my ( $class, $name ) = @_;
 
-dickyince, C<< <minnimince at googlemail.com> >>
+  my $self = ( name => $name, colour => _default_colour() );
 
-=head1 BUGS
+  bless $self, $class;
+}
 
-Please report any bugs or feature requests to C<bug-animal at rt.cpan.org>, or through
-the web interface at L<http://rt.cpan.org/NoAuth/ReportBug.html?Queue=Animal>.  I will be notified, and then you'll
-automatically be notified of progress on your bug as I make changes.
+=head2 _default_colour
 
+Returns the default colour of an animal
 
+=cut
 
+sub _default_colour {
+  return 'brown';
+}
 
-=head1 SUPPORT
+=head set_colour
 
-You can find documentation for this module with the perldoc command.
+Method to set the colour of a named animal
 
-    perldoc Animal
+=cut
 
+sub {
+  my $self = shift;
 
-You can also look for information at:
+  croak 'This is not a named animal' unless ref $self;
 
-=over 4
+  my $self->{colour} = shift;
+}
 
-=item * RT: CPAN's request tracker (report bugs here)
+=head colour
 
-L<http://rt.cpan.org/NoAuth/Bugs.html?Dist=Animal>
+Get the colour of an animal
 
-=item * AnnoCPAN: Annotated CPAN documentation
+=cut
 
-L<http://annocpan.org/dist/Animal>
+sub {
+  my $either = shift;
 
-=item * CPAN Ratings
+  ref $either ?
+    $either->{colour}
+    :
+    _default_colour;
```

Comment:
Since your methods are in a package, this method hasn't necessary been called yet so it needs an ampersand or parentheses so perl recognises it as a function name rather than a constant.